### PR TITLE
Adds verify-helm-docs.sh and update-helm-docs.sh scripts. Verify helm-docs during lint

### PR DESCRIPTION
**Pull request by JoshVanL**
/assign @SgtCoDFish 

This PR adds a scripts, run at `make lint`, which verifies that the helm chart `README.md` file is kept up to date.

An update script is also added which will update the helm chart `README.md`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -30,10 +30,14 @@ all: depend generate test build image ## runs test, build and image
...
-.PHONY: test
-lint:
+.PHONY: lint
+lint: depend helm-docs
 	./hack/verify-boilerplate.sh
 
+.PHONY: helm-docs
+helm-docs: # verify helm-docs
+	./hack/verify-helm-docs.sh
+
```
Comment (SgtCoDFish) [inline Makefile:40]: nit (non-blocking): Could we use `make verify-helm-docs` here? I think it'd be a little clearer that it's a verification step and that nothing is built from that make target.

**Review by SgtCoDFish — CHANGES_REQUESTED** [review]
I have a suggestion on how we're installing this tool, but I really like the idea of this PR and I think we should definitely have something like this in `cert-manager/trust`!

File: go.mod [hunk 2]
```diff
@@ -4,6 +4,7 @@ go 1.17
 
 require (
 	github.com/go-logr/logr v0.4.0
+	github.com/norwoodj/helm-docs v1.9.1
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply go.mod]: [helm-docs](https://github.com/norwoodj/helm-docs/releases) doesn't provide all of the target arches.

To resolve this I have added a separate `go.mod` for `hack/bin` so is no longer vendred at the top level.

Makefile binary deps are now built from that bin directory to use that `go.mod`

**Comment by JoshVanL** [comment]
Moved to https://github.com/cert-manager/trust/pull/28

/close

**Pull request closed without merging**